portalwire/portal_protocol_v1.go

```diff
@@ -85,17 +85,28 @@ func (a *AcceptV1) GetKeyLength() int {
 	return len(a.GetContentKeys())
 }
 
-func (p *PortalProtocol) getHighestVersion(node *enode.Node) (uint8, error) {
-	versions := &protocolVersions{}
-	err := node.Load(versions)
-	// key is not set, return the default version
-	if enr.IsNotFound(err) {
-		return p.currentVersions[0], nil
-	}
-	if err != nil {
-		return 0, err
+func (p *PortalProtocol) getOrStoreHighestVersion(node *enode.Node) (uint8, error) {
+	hcVersionValue, ok := p.versionsCache.Get(node)
+
+	if !ok {
```

Comment:
This `if-else` is better to reverse or just early return and not use `else`, since the else branch is very small.